app/jobs/daily_alert_mailer_job.rb

```diff
@@ -0,0 +1,37 @@
+class DailyAlertMailerJob < ActionMailer::DeliveryJob
+  EXPIRES_IN = 4.hours
+
+  before_enqueue do
```

Comment:
Difficult to spot the diff for the `after_queue` to `before_queue` fix, do we have test coverage for this case that will catch regressions?

Reply:
Good call. I rushed it a bit because it was breaking in testing. I'll see what I can do to catch the regressions

Reply:
I've added a test here that checks everything gets run in order https://github.com/dxw/teacher-vacancy-service/blob/cdeba62b9194fac725c9f8d773de45f40097593b/spec/jobs/daily_alert_mailer_job_spec.rb#L26-L33

Reply:
As discussed, we discovered that this meant the `job_id` was never being set. We thought it would work if it went into the `after_enqueue` block.

Reply:
Done this now. I've also added another spec the checks the job ID gets checked after the job is run:

https://github.com/dxw/teacher-vacancy-service/blob/c569fa9d670beae2e1c6745407b602b42d9e95dd/spec/jobs/daily_alert_mailer_job_spec.rb#L38-L47